### PR TITLE
Don't sync unnecessarily.

### PR DESCRIPTION
**Pull request by ironcamel**

### CODE REVIEW

**Comment by throughnothing** [comment]
Awesome.  Just add yourself to authors and I'll merge.

File: bin/notes [hunk 1]
```diff
@@ -39,33 +39,32 @@ sub pre_process {
...
+    sync( $c, pull_only => 1 ) if auto_sync();
 }
 
 sub post_process {
     my ( $c ) = @_;
     my $cmd = $c->cmd;
-    if( auto_sync() && grep { !/$cmd/ } qw( list show ) ) {
-        sync( $c, push_only => 1 );
+    if( $c->is_command($cmd) and not $cmd ~~ [qw( help list show sync )] ) {
+        sync( $c, push_only => 1 ) if auth_sync();
```
Comment (throughnothing) [inline bin/notes]: auth_sync() => auto_sync()

File: bin/notes [hunk 2]
```diff
@@ -39,33 +39,32 @@ sub pre_process {
...
-
-    unless ($c->is_command($cmd)) {
-        my @cmds = grep { /^$cmd/ } $c->commands;
-        $cmd = $cmds[0] if (@cmds == 1);
-    }
-
-    $c->execute( $cmd ) and return if $c->is_command( $cmd );
-    $c->execute( 'help' ) and return;
-}
+# TODO: This is broken. notes li will list stuff, then displays help.
```
Comment (throughnothing) [inline bin/notes]: This behavior doesn't happen to me on the current HEAD of master.

Comment (ironcamel) [inline bin/notes]: OK, I'll leave this alone in this PR.

**Pull request merged**